plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -727,22 +742,15 @@ public WriteMapping toWriteMapping(ConnectorSession session, Type type)
         }
 
         if (type instanceof TimestampType timestampType) {
-            if (timestampType.getPrecision() <= POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION) {
-                verify(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION);
-                return WriteMapping.longMapping(format("timestamp(%s)", timestampType.getPrecision()), PostgreSqlClient::shortTimestampWriteFunction);
-            }
-            verify(timestampType.getPrecision() > TimestampType.MAX_SHORT_PRECISION);
-            return WriteMapping.objectMapping(format("timestamp(%s)", POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION), longTimestampWriteFunction());
+            verify(timestampType.getPrecision() <= TimestampType.MAX_SHORT_PRECISION);
```

Comment:
How about timestamp with higher precision ?

Reply:
The engine takes care of it now, as you can see in the coercion tests?

Reply:
My bad. I thought POSTGRESQL_MAX_SUPPORTED_TIMESTAMP_PRECISION is more than 6.